src/views/AnalysisModule/AnalysisEditModal/index.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { useMemo, useState, useCallback } from 'react';
+import {
+    _cs,
+    randomString,
+    isNotDefined,
+    isDefined,
+} from '@togglecorp/fujs';
+import Faram, {
+    requiredCondition,
+    FaramActionElement,
+    FaramList,
+} from '@togglecorp/faram';
+
+import useRequest from '#utils/request';
+import Modal from '#dui/Modal';
+import Button from '#dui/Button';
+import TextInput from '#rsci/TextInput';
+import SelectInput from '#rsci/SelectInput';
+import Icon from '#rscg/Icon';
+import List from '#rscv/List';
+
+import { flatten } from '#utils/common';
+import { getMatrix1dToc, getMatrix2dToc } from '#utils/framework';
+import { notifyOnFailure } from '#utils/requestNotify';
+import {
+    MatrixTocElement,
+    MultiResponse,
+    FrameworkFields,
+    UserMini,
+} from '#typings';
+
+import _ts from '#ts';
+
+import PillarAnalysisRow from './PillarAnalysisRow';
+import styles from './styles.scss';
+
+const FaramButton = FaramActionElement(Button);
+
+const addAttribute = attributes => ([
+    ...attributes,
+    {
+        key: randomString(16),
+    },
+]);
+
+interface AnalysisEditModalProps {
+    className?: string;
+    onModalClose: () => void;
+    // TODO
+    value?: unknown;
+    projectId: number;
+}
+
+const analysisSchema = {
+    fields: {
+        title: [requiredCondition],
+        teamLead: [requiredCondition],
+        analysisPillar: [],
+    },
+};
+
+const userKeySelector = (u: UserMini) => u.id;
+const userLabelSelector = (u: UserMini) => u.displayName;
+
+const childrenSelector = (d: MatrixTocElement) => d.children;
+const analysisPillarKeySelector = d => d.key;
+
+function AnalysisEditModal(props: AnalysisEditModalProps) {
+    const {
+        className,
+        onModalClose,
+        value,
+        projectId,
+    } = props;
+
+    const [faramValues, setFaramValues] = useState(() => {
+        if (isNotDefined(value)) {
+            return undefined;
+        }
+        const newValue = {
+            ...value,
+            analysisPillar: value?.analysisPillar?.map(ap => ({
+                ...ap,
+                key: randomString(16),
+            })),
+        };
+        return newValue;
+    });
+    const [faramErrors, setFaramErrors] = useState();
+    const [pristine, setPristine] = useState(true);
+    const [bodyToSend, setBodyToSend] = useState(undefined);
```

Comment:
Any reason you passed undefined here as initial value for 'bodyToSend' but left it blank in 'faramErrors'? What's the difference?

Reply:
There is no difference.
I'll need to write better types for faramErrors as well.
I'll add a TODO.